src/test/java/io/confluent/connect/hdfs/TestWithMiniDFSCluster.java

```diff
@@ -63,10 +99,234 @@ private MiniDFSCluster createDFSCluster(Configuration conf) throws IOException {
     return cluster;
   }
 
-  @Override
-  protected Map<String, String> createProps() {
-    Map<String, String> props = super.createProps();
-    props.put(HdfsSinkConnectorConfig.HDFS_URL_CONFIG, url);
-    return props;
+  /**
+   * Return a list of new records starting at zero offset.
+   *
+   * @param size the number of records to return.
+   * @return the list of records.
+   */
+  protected List<SinkRecord> createSinkRecords(int size) {
+    return createSinkRecords(size, 0);
+  }
+
+  /**
+   * Return a list of new records starting at the given offset.
+   *
+   * @param size the number of records to return.
+   * @param startOffset the starting offset.
+   * @return the list of records.
+   */
+  protected List<SinkRecord> createSinkRecords(int size, long startOffset) {
+    return createSinkRecords(size, startOffset, Collections.singleton(new TopicPartition(TOPIC, PARTITION)));
+  }
+
+  /**
+   * Return a list of new records for a set of partitions, starting at the given offset in each partition.
+   *
+   * @param size the number of records to return.
+   * @param startOffset the starting offset.
+   * @param partitions the set of partitions to create records for.
+   * @return the list of records.
+   */
+  protected List<SinkRecord> createSinkRecords(int size, long startOffset, Set<TopicPartition> partitions) {
+    Schema schema = createSchema();
+    Struct record = createRecord(schema);
+    List<Struct> same = new ArrayList<>();
+    for (int i = 0; i < size; ++i) {
+      same.add(record);
+    }
+    return createSinkRecords(same, schema, startOffset, partitions);
+  }
+
+  protected List<SinkRecord> createSinkRecords(List<Struct> records, Schema schema) {
+    return createSinkRecords(records, schema, 0, Collections.singleton(new TopicPartition(TOPIC, PARTITION)));
+  }
+
+  protected List<SinkRecord> createSinkRecords(List<Struct> records, Schema schema, long startOffset,
+                                               Set<TopicPartition> partitions) {
+    String key = "key";
+    List<SinkRecord> sinkRecords = new ArrayList<>();
+    for (TopicPartition tp : partitions) {
+      long offset = startOffset;
+      for (Struct record : records) {
+        sinkRecords.add(new SinkRecord(TOPIC, tp.partition(), Schema.STRING_SCHEMA, key, schema, record, offset++));
+      }
+    }
+    return sinkRecords;
+  }
+
+  protected List<SinkRecord> createSinkRecordsNoVersion(int size, long startOffset) {
```

Comment:
Seems like there is probably some opportunity for test utilities in storage-common since a lot of these don't seem dependent on anything in this repo and look very familiar -- it's all the common data generation & validation code.